Make list/nav follow same patterns as top-bar/nav

justlevine opened the pull request:
When the `title` + `aria-label` were added to the `v2/list/nav` templates, introduced unnecessary complexity, and made the difference between these templates and `v2/list/top-bar/nav` templates more bothersome.

So I updated these templates to match the code patterns used in `v2/list/top-bar/nav`, mainly defining `$label` (with the `sprintf` and `tribe_get_event_label_plural` only being called once) at the top of the template, and then using `esc_attr( $label )`, everywhere else it's necessary.

This also let me move the nested `span` out of php and into the html itself.

**Note**:  I didnt touch the docblock `@vars` since they anyway don't match the files, and I didn't want to step on anyone's toes if its intentional.

I'm still not clear why there needs to be separate templates for `v2/list/nav` and `v2/list/top-bar/nav`, why `v2/day` and `v2/month` need their own nav templates instead of just using a singular nav template that uses the current view to decide if if the text should read 'Day', 'Month' or 'Events'... but if the template files can't be DRY, at least they can be consistent, and make it slightly easier for people to modify.

bordoni commented:
@justlevine thanks so much for submitting this Pull Request to us.

The only thing that we need to make sure we still keep intact is the code below unless you have a better way to maintain the behavior:

```
$events_label = '<span class="tribe-events-c-nav__next-label-plural tribe-common-a11y-visual-hide">' . tribe_get_event_label_plural() . '</span>';
echo wp_kses(
/* translators: %s: Event (plural or singular). */
	sprintf( $label, $events_label ),
	[ 'span' => [ 'class' => [] ] ]
);
```

The purpose of that is to allow hiding the "Events" label when on Mobile, so the navigation becomes smaller so we can prevent it from overflowing into two lines.

justlevine commented:
@bordoni - not sure I understand.

What I did was take the original `$events_label` and 
1. Move the `span` out of the `sprintf` and into the HTML markup. 
2. .Renamed `$events_label` to `$label` and moved it to the top of the file, so it matches the the top-bar/nav templates.

The resulting HTML output is exactly the same, it just only calls `tribe_get_event_label_plural()` and `sprintf` once, gets rid of the the need for the `wp_kses` and outputs the label using `esc_attr`.

Didn't do more than a cursory test since the generated html remains the same.

bordoni commented on src/views/v2/list/nav/next-disabled.php:
```diff
@@ -12,23 +12,21 @@
-					[ 'span' => [ 'class' => [] ] ]
-				);
-			?>
+			<span class="tribe-events-c-nav__next-label-plural tribe-common-a11y-visual-hide">
+				<?php echo esc_attr( $label ); ?>
+			</span>
```
```suggestion
			<?php
				$events_label = '<span class="tribe-events-c-nav__next-label-plural tribe-common-a11y-visual-hide">' . tribe_get_event_label_plural() . '</span>';
				echo wp_kses(
					/* translators: %s: Event (plural or singular). */
					sprintf( __( 'Next %1$s', 'the-events-calendar' ), $events_label ),
					[ 'span' => [ 'class' => [] ] ]
				);
			?>
```

bordoni commented on src/views/v2/list/nav/next.php:
```diff
@@ -14,25 +14,23 @@
-					[ 'span' => [ 'class' => [] ] ]
-				);
-			?>
+			<span class="tribe-events-c-nav__next-label-plural tribe-common-a11y-visual-hide">
+				<?php echo esc_attr( $label ); ?>
+			</span>
```
```suggestion
			<?php
				$events_label = '<span class="tribe-events-c-nav__next-label-plural tribe-common-a11y-visual-hide">' . tribe_get_event_label_plural() . '</span>';
				echo wp_kses(
					/* translators: %s: Event (plural or singular). */
					sprintf( __( 'Next %1$s', 'the-events-calendar' ), $events_label ),
					[ 'span' => [ 'class' => [] ] ]
				);
			?>
```

bordoni commented on src/views/v2/list/nav/prev-disabled.php:
```diff
@@ -12,23 +12,21 @@
-					[ 'span' => [ 'class' => [] ] ]
-				);
-			?>
+			<span class="tribe-events-c-nav__prev-label-plural tribe-common-a11y-visual-hide">
+				<?php echo esc_attr( $label ); ?>
+			</span>
```
```suggestion
			<?php
				$events_label = '<span class="tribe-events-c-nav__prev-label-plural tribe-common-a11y-visual-hide">' . tribe_get_event_label_plural() . '</span>';
				echo wp_kses(
					/* translators: %s: Event (plural or singular). */
					sprintf( __( 'Previous %1$s', 'the-events-calendar' ), $events_label ),
					[ 'span' => [ 'class' => [] ] ]
				);
			?>
```

bordoni commented on src/views/v2/list/nav/prev.php:
```diff
@@ -14,25 +14,23 @@
-					[ 'span' => [ 'class' => [] ] ]
-				);
-			?>
+			<span class="tribe-events-c-nav__prev-label-plural tribe-common-a11y-visual-hide">
+				<?php echo esc_attr( $label ); ?>
+			</span>
```
```suggestion
			<?php
				$events_label = '<span class="tribe-events-c-nav__prev-label-plural tribe-common-a11y-visual-hide">' . tribe_get_event_label_plural() . '</span>';
				echo wp_kses(
					/* translators: %s: Event (plural or singular). */
					sprintf( __( 'Previous %1$s', 'the-events-calendar' ), $events_label ),
					[ 'span' => [ 'class' => [] ] ]
				);
			?>
```

justlevine commented:
@bordoni 

Your changes made your earlier comment a clearer to me. Wasn't aware that it was the specific `tribe_get_event_label_plural` that needs to be wrapped for the mobile hidden to work. 

Two minor suggestions:
1. Perhaps there's a better name for `$events_label` to make it more clear how it differs from `$label` ? I'm sure yall have a style guide, but maybe `$events_label_wrapped`?

2. I _think_ the general trend with the v2 views was to move as much logic and var definitions to the top of the file, and out of the presentation. If that's true, perhaps move `$events_label` to the top as well? Something along the lines of 
Line 19: 
```php
/* translators: %s: Event (plural or singular). */
$label = sprintf( __( 'Previous %1$s', 'the-events-calendar' ), tribe_get_event_label_plural() );

/* translators: %s: Event (plural or singular). */
$events_label = sprintf( __( 'Previous %1$s', 'the-events-calendar'), '<span class="tribe-events-c-nav__prev-label-plural tribe-common-a11y-visual-hide">' . tribe_get_event_label_plural() . '</span>'); // Maybe rename $events_label to distinguish it from $label
```
Line 32:
```php
echo wp_kses( 
	$events_label, 
	[ 'span' => [ 'class' => [] ] ] 
);
```

bordoni commented:
Ohh yeah @justlevine that is a really good point possibly using the name `$events_mobile_friendly_label` would be the best variable name. That is super good point. Thanks for bringing it up. Do you want to get those changes commited to your branch or should it?

bordoni commented on src/views/v2/list/nav/next-disabled.php:
```diff
@@ -12,12 +12,15 @@
  */
+
+/* translators: %s: Event (plural or singular). */
+$label = sprintf( __( 'Next %1$s', 'the-events-calendar' ), tribe_get_event_label_plural() );
```
```suggestion
$label = sprintf( __( 'Next %1$s', 'the-events-calendar' ), tribe_get_event_label_plural() );

/* translators: %s: Event (plural or singular). */ 
$events_mobile_friendly_label = sprintf( __( 'Next %1$s', 'the-events-calendar' ), '<span class="tribe-events-c-nav__prev-label-plural tribe-common-a11y-visual-hide">' . tribe_get_event_label_plural() . '</span>' );
```

bordoni commented on src/views/v2/list/nav/next.php:
```diff
@@ -14,15 +14,18 @@
  */
+
+/* translators: %s: Event (plural or singular). */
+$label = sprintf( __( 'Next %1$s', 'the-events-calendar' ), tribe_get_event_label_plural() );
```
```suggestion
$label = sprintf( __( 'Next %1$s', 'the-events-calendar' ), tribe_get_event_label_plural() );

/* translators: %s: Event (plural or singular). */ 
$events_mobile_friendly_label = sprintf( __( 'Next %1$s', 'the-events-calendar' ), '<span class="tribe-events-c-nav__prev-label-plural tribe-common-a11y-visual-hide">' . tribe_get_event_label_plural() . '</span>' );
```

bordoni commented on src/views/v2/list/nav/prev-disabled.php:
```diff
@@ -12,12 +12,15 @@
  */
+
+/* translators: %s: Event (plural or singular). */
+$label = sprintf( __( 'Previous %1$s', 'the-events-calendar' ), tribe_get_event_label_plural() );
```
```suggestion
$label = sprintf( __( 'Previous %1$s', 'the-events-calendar' ), tribe_get_event_label_plural() );

/* translators: %s: Event (plural or singular). */ 
$events_mobile_friendly_label = sprintf( __( 'Previous %1$s', 'the-events-calendar' ), '<span class="tribe-events-c-nav__prev-label-plural tribe-common-a11y-visual-hide">' . tribe_get_event_label_plural() . '</span>' );
```

bordoni commented on src/views/v2/list/nav/prev.php:
```diff
@@ -14,15 +14,18 @@
  */
+
+/* translators: %s: Event (plural or singular). */
+$label = sprintf( __( 'Previous %1$s', 'the-events-calendar' ), tribe_get_event_label_plural() );
```
```suggestion
$label = sprintf( __( 'Previous %1$s', 'the-events-calendar' ), tribe_get_event_label_plural() );

/* translators: %s: Event (plural or singular). */ 
$events_mobile_friendly_label = sprintf( __( 'Previous %1$s', 'the-events-calendar' ), '<span class="tribe-events-c-nav__prev-label-plural tribe-common-a11y-visual-hide">' . tribe_get_event_label_plural() . '</span>' );
```
